### PR TITLE
Update small UG bugs based on PED

### LINKED ISSUE #280: [PE-D][Tester B] Missing details about corrupted data file
User of the app should be notified (somewhere in the UI) when the data file is corrupted. Currently, the data file is overwritten with an empty file, which might cause the user to be confused as to what is happening if all her data is deleted when the app is opened.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: RyanNgWH/ped#13

### PR DESCRIPTION
**Pull request by wapisai**
fix #280 #276 #273 #265 #263 #252 #250 #246

Fixed:
1. Make warning about corrupted data file more visible (#280)
2. Add in glossary and specify more about fields (#276)
3. Justify why clear command is long (#273)
4. Correct error message for delete n/NAME INDEX
5. Clarify error message for index of delete commmand
6. Fix typo in field constraints section (#252)
7. Fix typo in example add command in command summary (#250)
8. Clarify that indexes used in commands are based on the current (filtered/unfiltered) list (#246)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -38,6 +38,41 @@ command to run the application.<br>
...
+## Icon Legend
+
+| Icon | Meaning                                                                                                    |
+|------|------------------------------------------------------------------------------------------------------------|
+| 💡   | Tips, helpful suggestions and best practices to optimize the Realodex user experience experience.          |
+| ⚠️   | Warning for potential pitfalls or important considerations that could impact the Realodex user experience. |
+| 🗒️  | Additional Notes to keep track of in the features.                                                         |
+| 👀   | Upcoming Features to look out for. Stay tuned for exciting new features and enhancements in Realodex.      |
+
+## Glossary
```
Comment (jyztintan) [inline docs/UserGuide.md:63]: great!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -303,6 +341,8 @@ calculated by the number of days until their next birthday relative to the curre
 
 Clears all clients in Realodex.
 
+💡A longer command, `clearRealodex` is used so that users understand that this command **clears all entries in Realodex**, preventing potential confusion with the delete command and accidental clearing of all entries.
```
Comment (jyztintan) [inline docs/UserGuide.md:344]: yep

**Review by jyztintan — APPROVED** [review]
LGTM

**Review by UdhayaShan1 — APPROVED** [review]
LGTM, may add some stuff later, thanks!

File: docs/UserGuide.md (hunk 2 above)
Comment (appleraincoat) [inline docs/UserGuide.md:344]: thank you! 🔥

**Review by appleraincoat — APPROVED** [review]
LGTM

**Pull request merged**